spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/util/ByteBufferUtil.scala

```diff
@@ -0,0 +1,28 @@
+package com.datastax.spark.connector.util
+
+import java.nio.ByteBuffer
+
+object ByteBufferUtil {
+
+  /** Copies the remaining bytes of the buffer into the given array, starting from offset zero.
+    * The array must have capacity to store all of the remaining bytes of the buffer.
+    * The buffer's position remains untouched. */
+  def copyBuffer(src: ByteBuffer, dest: Array[Byte]): Array[Byte] = {
+    if (src.hasArray) {
+      val length = src.remaining
+      val offset =  src.arrayOffset + src.position
+      System.arraycopy(src.array, offset, dest, 0, length)
+    } else {
+      src.duplicate.get(dest)
+    }
+    dest
+  }
```

Comment:
I see what you're trying to achieve here, but now you introduced another problem that the dest array is not used at all in one of the branches.